### ISSUE TITLE
[Linux] Crash on `layer-toggle` to current layer

### ISSUE THREAD

**Issue by MaxGyver83** [david-janssen/kmonad#119]
I just noticed that when I run [my own config](https://github.com/david-janssen/kmonad/blob/master/keymap/user/MaxGyver83/vou.kbd), and I hold the left shift key and then press the right shift key, KMonad crashes.

The reason is obvious: The left shift key toggles layer 2. The right shift key on layer 2 also toggles layer 2. I think, I should put an `XX` for both shift keys on layer 2.

Maybe it's worth catching this crash (although it was caused by a bad config).

The debug output:

```
Opening KeySink
Registering Uinput device
Running UinputSink command: "/bin/sleep 1 && /bin/setxkbmap us -option compose:ralt"
Opening KeySource
Initiating ioctl grab
Launching process: emitter_proc

--------------------------------------------------------------------------------
Received event: Press <sft>
Running hooks
Pushed layer to stack: level2
 1. level2
Base-layer: level1

Registering untimed hook: 1

--------------------------------------------------------------------------------
Received event: Press <rsft>
Running hooks
Registering untimed hook: 2
Pushed layer to stack: level2
 1. level2
Base-layer: level1

Registering untimed hook: 3

--------------------------------------------------------------------------------
Received event: Release <rsft>
Running hooks
Registering untimed hook: 4
Popped layer from stack: level2
Base-layer: level1


--------------------------------------------------------------------------------
Received event: Release <sft>
Running hooks
Closing process: emitter_proc
Closing KeySource
Releasing ioctl grab
Closing KeySink
Unregistering Uinput device
Closing Uinput device file
kmonad: LayerNotOnStack "level2"
```

**Comment by david-janssen** [comment]
Goooood catch. This definitely needs fixing. Core bugs go to the top of the pile.

**Comment by david-janssen** [comment]
Fixed as of: 0e7feb5257007d4068f034dee327d1c64bb3c7e6

The first release will pop the layer, any subsequent releases (i.e. of other buttons that *also* added the layer) will generate warnings, because I do think it is advisable to avoid these types of situations altogether, but won't crash KMonad.

**Comment by MaxGyver83** [comment]
Awesome! Thank you! :+1:
